crates/cli/src/opts/dependency.rs

```diff
@@ -360,4 +370,26 @@ mod tests {
         let dep: Dependency = "org-git12345678@github.com:my-org/my-repo.git".parse().unwrap();
         assert_eq!(dep.url.unwrap(), "https://github.com/my-org/my-repo");
     }
+
+    #[test]
+    fn can_parse_with_explicit_ref_type() {
+        let dep = Dependency::from_str("smartcontractkit/ccip@tag=contracts-ccip/v1.2.1").unwrap();
+        assert_eq!(dep.name, "ccip");
+        assert_eq!(dep.url, Some("https://github.com/smartcontractkit/ccip".to_string()));
+        assert_eq!(dep.tag, Some("contracts-ccip/v1.2.1".to_string()));
+        assert_eq!(dep.alias, None);
+
+        let dep =
+            Dependency::from_str("smartcontractkit/ccip@branch=contracts-ccip/v1.2.1").unwrap();
+        assert_eq!(dep.name, "ccip");
+        assert_eq!(dep.url, Some("https://github.com/smartcontractkit/ccip".to_string()));
+        assert_eq!(dep.tag, Some("contracts-ccip/v1.2.1".to_string()));
```

Comment:
would it already be desired to differentiate between tags and branches here by storing them in a different field? or would that be a follow-up?

https://github.com/foundry-rs/foundry/issues/7225

Reply:
Ah, good point, I think could be followed up, would that work?

Reply:
Yes we can do that as a follow up